dts: bindings: comparator: nordic: Change ext analog inputs type to int

jaz1-nordic opened the pull request:
Unify external analog inputs type to be consistent in COMP, LPCOMP and SAADC nordic drivers.

jaz1-nordic committed:
dts: bindings: comparator: nordic: Change inputs type to int

Unify external analog inputs type to be consistent
in COMP, LPCOMP and SAADC nordic drivers.

Signed-off-by: Jakub Zymelka <jakub.zymelka@nordicsemi.no>

jaz1-nordic committed:
drivers comparator: nordic: Align drivers to changed analog input types

External analog input types changed from `string` to `int`.

Signed-off-by: Jakub Zymelka <jakub.zymelka@nordicsemi.no>

jaz1-nordic committed:
tests: comparator: nordic: Align boards overlays for changed input types

Align samples and tests boards overlays
for changed comparator input types.

Signed-off-by: Jakub Zymelka <jakub.zymelka@nordicsemi.no>

decsny commented:
if this gets approved, will need to update migration guide

carlescufi commented:
@bjarki-andreasen and @anangl please take a look

bjarki-andreasen commented on include/zephyr/drivers/comparator/nrf_lpcomp.h:
```diff
@@ -84,9 +56,9 @@ enum comp_nrf_lpcomp_refsel {
 struct comp_nrf_lpcomp_config {
 	/** Positive input selection */
-	enum comp_nrf_lpcomp_psel psel;
+	uint8_t psel;
```
This change requires including the bindings which define the psel and refsel, and extending the documentation both the defines in the bindings, similar to the comments in the enums. So, add
```
/** AIN0 external input */
```
the the defines in the bindings file and extend the psel docs to mention where the defines are from
```
/** Positive input selection  defined by the NRF_LPCOMP_ defines */
```
and add the required header above to the header file instead if the source file as its required by the user to use the API
```
#include <zephyr/dt-bindings/comparator/nrf-lpcomp.h>
```

bjarki-andreasen replied:
Since we loose the enums, we loose the strict types, so we need to be extra careful to document exactly what integer goes where, since anything could be passed to the API

jaz1-nordic committed:
doc: update migration guide for Nordic comparator changes

Migration guide is updated to report the changes that has been introduced
in the nordic,nrf-comp and nordic,nrf-lpcomp bindings.
`psel` and `extrefsel` properties type has been changed to integer.

Signed-off-by: Jakub Zymelka <jakub.zymelka@nordicsemi.no>

bjarki-andreasen reviewed (changes requested):
One typo otherwise looks good (also there is one too long line)
